File: src/main/java/io/github/gravetii/scene/start/StartScene.java
package io.github.gravetii.scene.start;

import io.github.gravetii.scene.FxDimensions;
import io.github.gravetii.scene.FxScene;
import io.github.gravetii.scene.menu.MenuBarComponent;
import javafx.geometry.Dimension2D;
import javafx.stage.Stage;

import java.util.Optional;

public class StartScene extends FxScene {
  private MenuBarComponent menuBarComponent;
  private StartImageComponent startImageComponent;
  private StartControlsComponent startControlsComponent;

  public StartScene(Stage stage) throws Exception {
    super(stage);
    this.menuBarComponent = new MenuBarComponent(this.stage, this.root);
    this.startImageComponent = new StartImageComponent(this.root);
    this.startControlsComponent = new StartControlsComponent(menuBarComponent);
  }

  @Override
  protected void build() {
    this.showTop(this.menuBarComponent)
        .showCenter(this.startImageComponent)
        .showRight(this.startControlsComponent);
  }

  @Override
  public String title() {
    return "-WORDAGAM-";
  }

  @Override
  protected Optional<FxDimensions> preferredDimensions() {
    FxDimensions dimensions =
        new FxDimensions(
            new Dimension2D(960, 630), new Dimension2D(960, 630), new Dimension2D(1100, 732));
    return Optional.of(dimensions);
  }
}

File: src/main/kotlin/io/github/gravetii/scene/start/StartScene.kt
package io.github.gravetii.scene.start

import io.github.gravetii.scene.FxDimensions
import io.github.gravetii.scene.FxScene
import io.github.gravetii.scene.menu.MenuBarComponent
import javafx.geometry.Dimension2D
import javafx.stage.Stage

class StartScene(stage: Stage) : FxScene(stage) {

    private val menuBarComponent = MenuBarComponent(stage, root)
    private val startImageComponent = StartImageComponent(root)
    private val startControlsComponent = StartControlsComponent(menuBarComponent)

    override fun build() {
        showTop(menuBarComponent)
        showCenter(startImageComponent)
        showRight(startControlsComponent)
    }

    override fun title() = "-WORDAGAM-"

    override fun preferredDimensions(): FxDimensions {
        val def = Dimension2D(960.0, 630.0)
        val min = Dimension2D(960.0, 630.0)
        val max = Dimension2D(1100.0, 732.0)
        return FxDimensions(def, min, max)
    }

}